### PR TITLE
gamescope: 3.12.5 -> 3.13.19, refactor

### PR DESCRIPTION
**Pull request by K900**
## Description of changes

This mostly brings in changes from Jovian, but also splits the dependencies between the WSI layer and the Gamescope executable, which simplifies i686 layer builds a whole lot.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by PedroHLC — COMMENTED** [review]
Really cool changes, can we make joshShaders overridable somehow?

**Comment by K900** [comment]
It'll also pick up shaders from ~/.reshade or something like that, that's just the default set.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -1812,7 +1812,13 @@ with pkgs;
...
   gamescope = callPackage ../applications/window-managers/gamescope {
-    wlroots = wlroots_0_16;
+    enableExecutable = true;
+    enableWsi = false;
+  };
+
+  gamescope-wsi = callPackage ../applications/window-managers/gamescope {
+    enableExecutable = false;
+    enableWsi = true;
   };
```
[the remark this answers is not in the document]
Comment (K900) [reply pkgs/top-level/all-packages.nix:1822]: It's currently the latest, we can always pin if necessary when 0.18 is released.

File: pkgs/applications/window-managers/gamescope/default.nix [hunk 2]
```diff
@@ -30,16 +26,18 @@
...
     owner = "Joshua-Ashton";
-    repo = "vkroots";
-    rev = "26757103dde8133bab432d172b8841df6bb48155";
-    hash = "sha256-eet+FMRO2aBQJcCPOKNKGuQv5oDIrgdVPRO00c5gkL0=";
+    repo = "GamescopeShaders";
+    rev = "v0.1";
+    hash = "sha256-gR1AeAHV/Kn4ntiEDUSPxASLMFusV6hgSGrTbMCBUZA=";
   };
 in
 stdenv.mkDerivation {
```
[the remark this answers is not in the document]
Comment (K900) [reply pkgs/applications/window-managers/gamescope/default.nix:43]: Done.

**Comment by Scrumplex** [comment]
What would happen if gamescope bumped its minimum wlroots version while we are still at 0.17.0? I assume it would just use the submodule now. This would be something we would need to pay attention to when updating this package in the future. Maybe we can stop Meson from using the wlroots submodule altogether?

**Comment by K900** [comment]
It would use the submodule, which will most likely fail because we don't have all the wlroots dependencies in the environment. I don't know if this is really a problem? We could go back to manually fetching all the submodules we know we want to vendor, I guess, but that also doesn't feel great.

**Comment by K900** [comment]
Also added myself as maintainer since we're already tracking this for Jovian.

**Comment by Scrumplex** [comment]
> We could go back to manually fetching all the submodules we know we want to vendor, I guess, but that also doesn't feel great.

Yeah, it wasn't great, really. We could also work on packaging the remaining submodules and just override their versions in this derivation.

vkroots is a low-hanging fruit here. reshade doesn't have CMake configs or pkg-config manifests, so we would be stuck with using the vendored version. Using our packaged SPIRV-Headers would need some changes in Gamescope.

**Comment by K900** [comment]
vkroots still changes pretty often, and gamescope is the only real consumer of it, so I don't really see the point.

**Comment by Scrumplex** [comment]
Result of `nixpkgs-review pr 280465` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>bottles</li>
    <li>bottles-unwrapped</li>
    <li>gamescope</li>
    <li>gamescope-wsi</li>
  </ul>
</details>

**Review by Scrumplex — APPROVED** [review]

**Comment by K900** [comment]
I'm also testing this on my Steam Deck with https://github.com/Jovian-Experiments/Jovian-NixOS/pull/265

**Pull request merged**